File: GXpert/GXpert.Web/Modules/LiveSessions/LiveSession/LiveSessionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.LiveSessions.Columns;

[ColumnsScript("LiveSessions.LiveSession")]
[BasedOnRow(typeof(LiveSessionRow), CheckNames = true)]
public class LiveSessionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string MeetingId { get; set; }
    public short LiveProvider { get; set; }
    public string Password { get; set; }
    public string Secret { get; set; }
    public string LiveAdditionalInfo { get; set; }
    public string SearchTags { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/LiveSessions/LiveSessionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ELiveProvider } from "../Web/Enums.ELiveProvider";
import { LiveSessionRow } from "./LiveSessionRow";

export interface LiveSessionColumns {
    Id: Column<LiveSessionRow>;
    MeetingId: Column<LiveSessionRow>;
    LiveProvider: Column<LiveSessionRow>;
    Password: Column<LiveSessionRow>;
    Secret: Column<LiveSessionRow>;
    LiveAdditionalInfo: Column<LiveSessionRow>;
    SearchTags: Column<LiveSessionRow>;
}

export class LiveSessionColumns extends ColumnsBase<LiveSessionRow> {
    static readonly columnsKey = 'LiveSessions.LiveSession';
    static readonly Fields = fieldsProxy<LiveSessionColumns>();
}

[ELiveProvider]; // referenced types